### PR TITLE
Update the version of LTP and prerequisites of test

### PR DESCRIPTION
**Pull request by xian123**
Added dependency packages and configuration steps to resolve LTP test suites failures.
Updated LTP version to 20190115

### CODE REVIEW

File: Testscripts/Linux/Linux-Test-Project-Tests.sh [hunk 1]
```diff
@@ -45,24 +48,38 @@ install_package "${common_packages[@]}"
...
+        install_epel
+        rpm_packages=(git db4-utils libaio-devel libattr libcap-devel libdb ntp)
         install_package "${rpm_packages[@]}"
         ;;
     *)
         LogMsg "Unknown distro $DISTRO, continuing to try for RPM installation"
         ;;
 esac
 
+rt_runtime_us=$(cat /sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us)
```
Comment (chvalean) [inline Testscripts/Linux/Linux-Test-Project-Tests.sh:73]: please add a comment for which LTP test this is required so we don't loose track

File: Testscripts/Windows/Run-LtpSuite.ps1 [hunk 2]
```diff
@@ -32,7 +32,13 @@ function Main {
         -Username $user -password $password `
         -TestName $currentTestData.testName
 
-    $filesTocopy = "{0}/${LTP_RESULTS}, {0}/${LTP_OUPUT}" -f @("/home/${username}")
+    # The LTP log will be placed under /root on SUSE and RedHat when running TEST_SCRIPT with runAsSudo
+    # The NULL.log makes sure cp always work on all distros
```
Comment (LiliDeng) [inline Testscripts/Windows/Run-LtpSuite.ps1:36]: hit test failure against ubuntu, please fix it.

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/Run-LtpSuite.ps1 (hunk 2 above)
Comment (LiliDeng) [reply Testscripts/Windows/Run-LtpSuite.ps1:36]: not caused by this reason, seems execute time is not enough sometimes

**Comment by juhlee-microsoft** [comment]
Please add description

File: Testscripts/Linux/Linux-Test-Project-Tests.sh [hunk 3]
```diff
@@ -45,28 +48,48 @@ install_package "${common_packages[@]}"
...
         ;;
     *)
         LogMsg "Unknown distro $DISTRO, continuing to try for RPM installation"
         ;;
 esac
 
+# Some CPU time is assigned to set real-time scheduler and it affects all cgroup test cases.
+rt_runtime_us=$(cat /sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us)
+if [ $rt_runtime_us -eq 0 ]; then
+    echo "950000" > /sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/Linux-Test-Project-Tests.sh:79]: what represent of 950000?

File: Testscripts/Windows/Run-LtpSuite.ps1 [hunk 4]
```diff
@@ -25,22 +25,35 @@ function Main {
...
         -files $filesTocopy
 
-    $LogPath = Join-Path $LogDir $LTP_RESULTS
-    $content = Get-Content $LogPath
-
-    $content | ForEach-Object {
+    $statusLogPath = Join-Path $LogDir "state.txt"
+    $currentResult = Get-Content $statusLogPath
+    if (($currentResult -imatch "TestAborted") -or ($currentResult -imatch "TestRunning")) {
+            Write-LogErr "Test aborted. Last known status : $currentResult"
```
Comment (juhlee-microsoft) [inline Testscripts/Windows/Run-LtpSuite.ps1]: check indentation

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

**Comment by LiliDeng** [comment]
Before this fix, most of run not finish in real meaning, it just stop after timeout, so we extend the time, and check the state.

```
Test Run On           : 04/29/2019 06:56:04
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:3:8

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 LTP                  LINUX-TEST-PROJECT-TESTS-FULL-RUN                                                 FAIL                185.1 
	fanotify09 : FAIL 
	quotactl01 : FAIL 
	quota_remount_test01 : FAIL 
	memcg_max_usage_in_bytes : FAIL 
	memcg_stat : FAIL 
	memcg_use_hierarchy : FAIL 
	memcg_usage_in_bytes : FAIL 
	memcg_stress : FAIL 
	cpuset_hotplug : FAIL 
	ar_sh : FAIL 
	block_dev : FAIL 
	getaddrinfo_01 : FAIL 
	cve-2017-17053 : FAIL 

[LISAv2 Test Results Summary]
Test Run On           : 04/29/2019 06:56:39
ARM Image Under Test  : SUSE : SLES : 15 : Latest
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:3:20

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 LTP                  LINUX-TEST-PROJECT-TESTS-FULL-RUN                                                 FAIL                195.6 
	inotify07 : FAIL 
	inotify08 : FAIL 
	memcg_failcnt : FAIL 
	memcg_max_usage_in_bytes : FAIL 
	memcg_stat : FAIL 
	memcg_use_hierarchy : FAIL 
	memcg_usage_in_bytes : FAIL 
	cgroup_fj_function_cpu : FAIL 
	cgroup_fj_function_cpuacct : FAIL 
	cgroup_fj_stress_cpu_2_2_none : FAIL 
	cgroup_fj_stress_cpu_3_3_none : FAIL 
	cgroup_fj_stress_cpu_4_4_none : FAIL 
	cgroup_fj_stress_cpu_2_9_none : FAIL 
	cgroup_fj_stress_cpu_10_3_none : FAIL 
	cgroup_fj_stress_cpu_1_200_none : FAIL 
	cgroup_fj_stress_cpu_200_1_none : FAIL 
	cgroup_fj_stress_cpu_2_2_one : FAIL 
	cgroup_fj_stress_cpu_3_3_one : FAIL 
	cgroup_fj_stress_cpu_4_4_one : FAIL 
	cgroup_fj_stress_cpu_2_9_one : FAIL 
	cgroup_fj_stress_cpu_10_3_one : FAIL 
	cgroup_fj_stress_cpu_1_200_one : FAIL 
	cgroup_fj_stress_cpu_200_1_one : FAIL 
	cgroup_fj_stress_cpu_2_2_each : FAIL 
	cgroup_fj_stress_cpu_3_3_each : FAIL 
	cgroup_fj_stress_cpu_4_4_each : FAIL 
	cgroup_fj_stress_cpu_2_9_each : FAIL 
	cgroup_fj_stress_cpu_10_3_each : FAIL 
	cgroup_fj_stress_cpu_1_200_each : FAIL 
	cgroup_fj_stress_cpu_200_1_each : FAIL 
	cgroup_fj_stress_cpuacct_2_2_none : FAIL 
	cgroup_fj_stress_cpuacct_3_3_none : FAIL 
	cgroup_fj_stress_cpuacct_4_4_none : FAIL 
	cgroup_fj_stress_cpuacct_2_9_none : FAIL 
	cgroup_fj_stress_cpuacct_10_3_none : FAIL 
	cgroup_fj_stress_cpuacct_1_200_none : FAIL 
	cgroup_fj_stress_cpuacct_200_1_none : FAIL 
	cgroup_fj_stress_cpuacct_2_2_one : FAIL 
	cgroup_fj_stress_cpuacct_3_3_one : FAIL 
	cgroup_fj_stress_cpuacct_4_4_one : FAIL 
	cgroup_fj_stress_cpuacct_2_9_one : FAIL 
	cgroup_fj_stress_cpuacct_10_3_one : FAIL 
	cgroup_fj_stress_cpuacct_1_200_one : FAIL 
	cgroup_fj_stress_cpuacct_200_1_one : FAIL 
	cgroup_fj_stress_cpuacct_2_2_each : FAIL 
	cgroup_fj_stress_cpuacct_3_3_each : FAIL 
	cgroup_fj_stress_cpuacct_4_4_each : FAIL 
	cgroup_fj_stress_cpuacct_2_9_each : FAIL 
	cgroup_fj_stress_cpuacct_10_3_each : FAIL 
	cgroup_fj_stress_cpuacct_1_200_each : FAIL 
	cgroup_fj_stress_cpuacct_200_1_each : FAIL 
	cpuacct_1_1 : FAIL 
	cpuacct_1_10 : FAIL 
	cpuacct_10_10 : FAIL 
	cpuacct_1_100 : FAIL 
	cpuacct_100_1 : FAIL 
	cpuacct_100_100 : FAIL 
	cpuset_hotplug : FAIL 
	cpuhotplug02 : FAIL 
	cpuhotplug03 : FAIL 
	cpuhotplug04 : FAIL 
	cpuhotplug06 : FAIL 
	getaddrinfo_01 : FAIL 
```

File: Testscripts/Linux/Linux-Test-Project-Tests.sh (hunk 3 above)
Comment (xian123) [reply Testscripts/Linux/Linux-Test-Project-Tests.sh:79]: The values for rt_period_us and rt_runtime_us (950000us or 0.95s), this gives 0.05s to be used by non-RT tasks.

Comment (juhlee-microsoft) [reply Testscripts/Linux/Linux-Test-Project-Tests.sh:79]: Then, please add it in comment.

File: Testscripts/Windows/Run-LtpSuite.ps1 [hunk 5]
```diff
@@ -25,22 +25,35 @@ function Main {
...
+    $currentResult = Get-Content $statusLogPath
+    if (($currentResult -imatch "TestAborted") -or ($currentResult -imatch "TestRunning")) {
+        Write-LogErr "Test aborted. Last known status : $currentResult"
+        $resultArr += "Aborted"
+        $CurrentTestResult.TestSummary += New-ResultSummary -testResult $currentResult -metaData $metaData `
+            -checkValues "PASS,FAIL,ABORTED" -testName $CurrentTestData.testName
+    } else {
+        $LogPath = Join-Path $LogDir $LTP_RESULTS
+        $content = Get-Content $LogPath
+        $content | ForEach-Object {
```
Comment (juhlee-microsoft) [inline Testscripts/Windows/Run-LtpSuite.ps1:56]: please align the line number 57 along with 56.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

**Comment by xian123** [comment]
Test Run On           : 04/30/2019 03:39:59
ARM Image Under Test  : SUSE : SLES-Standard : 15 : Latest
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:3:17

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 LTP                  LINUX-TEST-PROJECT-TESTS-FULL-RUN                                                 FAIL               193.58
        inotify07 : FAIL
        inotify08 : FAIL
        memcg_max_usage_in_bytes : FAIL
        memcg_stat : FAIL
        memcg_use_hierarchy : FAIL
        memcg_usage_in_bytes : FAIL
        cpuset_hotplug : FAIL
        cpuhotplug02 : FAIL
        cpuhotplug03 : FAIL
        cpuhotplug04 : FAIL
        cpuhotplug06 : FAIL
        getaddrinfo_01 : FAIL

File: Testscripts/Linux/Linux-Test-Project-Tests.sh [hunk 6]
```diff
@@ -45,28 +48,54 @@ install_package "${common_packages[@]}"
 case $DISTRO in
     "suse"*)
         suse_packages=(git-core db48-utils libaio-devel libattr1 \
-            libcap-progs libdb-4_8 perl-BerkeleyDB)
+            libcap-progs libdb-4_8 perl-BerkeleyDB ntp)
```
Comment (chvalean) [inline Testscripts/Linux/Linux-Test-Project-Tests.sh]: ntp looks to be a common package on all distros, please add it to line 44

**Review by chvalean — APPROVED** [review]

**Review by LiliDeng — APPROVED** [review]

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**